pkg/kubelet/kubelet.go

```diff
@@ -1088,6 +1089,7 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 				}
 				glog.Infof("pod %q container %q is unhealthy. Container will be killed and re-created.", podFullName, container.Name, live)
 			} else {
+				podChanged = true
```

Comment:
Ugh, there is an escape here - if a container adds/removes/changes a host port, we won't restart the network container.  I'm not too worried about that by itself, but are there any other aspects of a single container that might require the pod itself to be recreated?